packages/workflow/src/Cron.ts

```diff
@@ -0,0 +1,68 @@
+interface BaseTriggerTime<T extends string> {
+	mode: T;
+}
+
+type CronExpression = string;
+interface CustomTrigger extends BaseTriggerTime<'custom'> {
+	cronExpression: CronExpression;
+}
+
+interface EveryXTrigger<U extends string> extends BaseTriggerTime<'everyX'> {
+	unit: U;
+	value: number;
+}
+
+type EveryMinuteTrigger = BaseTriggerTime<'everyMinute'>;
+type EveryXMinutesTrigger = EveryXTrigger<'minutes'>;
+
+interface EveryHourTrigger extends BaseTriggerTime<'everyHour'> {
+	minute: number; // 0 - 59
+}
+type EveryXHoursTrigger = EveryXTrigger<'hours'>;
+
+interface EveryDayTrigger extends BaseTriggerTime<'everyDay'> {
```

Comment:
Overkill to have ranges?
```ts
type MakeRange<
    MaxExclusive extends number,
    Collector extends readonly number[] = [],
    > = Collector['length'] extends MaxExclusive
        ? Collector[number]
        : MakeRange<MaxExclusive, [...Collector, Collector['length']]>;

type HourOfDay = MakeRange<24>;
```

Reply:
I thought about it, and it did feel like an overkill. makes me wish for native range types in JS.